accounts/rpcserver.go

```diff
@@ -53,6 +53,10 @@ func (s *RPCServer) CreateAccount(ctx context.Context,
 	log.Infof("[createaccount] label=%v, balance=%d, expiration=%d",
 		req.Label, req.AccountBalance, req.ExpirationDate)
 
+	if !s.service.IsRunning() {
+		return nil, ErrAccountServiceDisabled
+	}
+
```

Comment:
cant we do a more generic thing where, similar to the existing sub-server manager, we have have a lit-system manager or something (eventually both the sub-server manager and lit-system manager could be merged or abstracted away into one `subserver manager`) and then we only need to check once in the `checkSubSystemStarted` `rpcProxy` method if the accounts service is up and ready. 

Then the account service itself is just responsible for signalling/reporting back that it is currently not ready for requests and then we dont have to repeat this `s.service.IsRunning` check everywhere here.

Reply:
Interested to hear what you guys think though cause perhaps there would be danger in doing it this way since it is harder to keep things atomic since we then depend on an external system to work correctly etc plus we would need to think about the case where accounts wants to be disabled but then Lit lets through just one more request in the mean time before the "disable" is noted. 

Another (perhaps more simple) suggestion would be to allow lit subservers like the accounts server to provide grpc Unary and Stream interceptors that it expects Lit to register with any gRPC server that it spins up that would give access to accounts. These interceptors can then do the "are we ready for a call" check. But perhaps this also has the "call might slip through" issue. 

Lemme know what you guys think

Reply:
Interested to hear if you think we can make your suggestion work for the issues I explained in https://github.com/lightninglabs/lightning-terminal/pull/642#issuecomment-1729293200 :)!